src/toolkits/style_transfer/style_transfer_model_definition.cpp

```diff
@@ -33,12 +35,20 @@ constexpr float LOWER_BOUND = -0.07;
 constexpr float UPPER_BOUND = 0.07; 
 
 // TODO: refactor code to be more readable with loops
-void define_resnet(model_spec& nn_spec, size_t num_styles, int random_seed=0) {
-  std::mt19937 random_engine;
-  std::seed_seq seed_seq{random_seed};
-  random_engine = std::mt19937(seed_seq);
-
-  auto initializer = uniform_weight_initializer(LOWER_BOUND, UPPER_BOUND, &random_engine);
+void define_resnet(model_spec& nn_spec, size_t num_styles, bool initialize=false, int random_seed=0) {
+  weight_initializer initializer;
+
+  // This is to make sure that when the uniform initialization is not needed extra work is avoided
+  if (initialize) {
+    std::mt19937 random_engine;
+    std::seed_seq seed_seq{random_seed};
+    random_engine = std::mt19937(seed_seq);
+
+    initializer = uniform_weight_initializer(LOWER_BOUND, UPPER_BOUND, &random_engine);
```

Comment:
I believe `&random_engine` will become a dangling pointer once `random_engine` goes out scope after this line. I suspect you will need to move `random_engine` outside the scope of this `if` statement to obtain well-defined behavior

Reply:
> @nickjong
> 
> > What is the relationship between this PR and #2874 ?
> 
> This fixes the waviness users were experiencing. The pre-trained model let's the users get faster models but this is more likely to introduce artifacts into the stylization. The uniform initialization let's the user train a higher quality model but with a higher number of iterations. It resolves (7.) of the issue

Are we exposing both pre trained weights as well as just the uniform initialization. I think if we are exposing it would be good for the users to actually have this literature somewhere. I don't think it is intuitive and I think we should put it in our userguide. 

I also think maybe in another PR you should expose the random_seed parameter. I think it would be great if users could generate reproducible models.

Reply:
I propose trying to have all the deep-learning toolkits controlled by random seeds for 6.2...